### ISSUE TITLE
Workspaces other than active one don't change their colour scheme

### ISSUE THREAD

**Issue by gegoune** [wez/wezterm#3328]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

20230319-202034-51d7b28f

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Changing colour scheme in response to change in system wide setting does apply changes to single (current) workspace only.

### To Reproduce

1. Configure Wezterm as per reproducible configuration below or as per example configuration in https://wezfurlong.org/wezterm/config/lua/window/get_appearance.html#windowget_appearance
2. Open at least one more Workspace
3. Use MacOS' functionality to switch to different than current (dark/light) mode

Observe colour scheme change for active workspace

1. Change to another workspace

Observe that new colour scheme didn't get applied here.

### Configuration

```lua
local wezterm = require 'wezterm'

local colors, _ = wezterm.color.load_scheme(
  (wezterm.home_dir .. '/.local/share/nvim/site/pack/packer/start/zenbones.nvim/extras/wezterm/Zenwritten_%s.toml'):format(
    wezterm.gui.get_appearance():lower()
  )
)

return { colors = colors }
```

### Expected Behavior

All windows/panes/workspaces to receive updated configuration and change their colour scheme accordingly.

Instead only current workspace gets new colour scheme applied.

### Anything else?

Potentially related to https://github.com/wez/wezterm/issues/3259

**Comment by wez** [comment]
I can't reproduce this. Are there steps missing from your reproduction instructions?

**Comment by wez** [comment]
This is the config I used for testing:

```lua
local wezterm = require 'wezterm'

function scheme_for_appearance(appearance)
  if appearance:find 'Dark' then
    return 'Builtin Solarized Dark'
  else
    return 'Builtin Solarized Light'
  end
end

return {
  color_scheme = scheme_for_appearance(wezterm.gui.get_appearance()),
}
```

**Comment by gegoune** [comment]
Thank you for looking into it. Surprisingly I cannot replicate it now as well, but just have updated to latest nightly. I wonder if that is because I just restarted Wezterm or there is no issue there no more. I can wait until automatic change at sunset to see if using terminal and having content in windows make any difference. 

I will keep issue open for now until I conduct bit more investigation, but since we both can't reproduce it right now I understand if you will close it.

**Comment by gegoune** [comment]
I have managed to trigger the issue. It required few changes between light/dark settings in operating system, switching between workspaces handful of times before colour schemes stopped changing for not current workspace.

I know it's very vague and I am not expecting it would push this issue forward, just wanted to update you and perhaps point towards bit more convoluted reproduction required. I will try to see if I can find concise pattern which results in described issue. Appreciate your patience here, thank you!

**Comment by gegoune** [comment]
Hey Wez, I manage to record a video of an issue.

Steps to reproduce:
- start Western and create another workspace, switch between workspaces once or twice
- start `nvim --clean` - not sure if that's required, don't have vim to test it unfortunately, change to other workspace
- change theme in operating system
- change workspace and notice how it's colorscheme is not updated.

I have used your configuration from https://github.com/wez/wezterm/issues/3328#issuecomment-1480638385 to test.
![switching](https://user-images.githubusercontent.com/69750637/228290677-23e9ce07-623e-4d47-8eae-7c1bc394892d.gif)



I have added some key mappings to ease with changing workspaces, other than that configuration is as in linked response.
I am not sure why Neovim is somewhat crucial in replicating that issue - this is the most reproducible scenario I managed to come up with, but I am pretty sure it happens even without Neovi involved.

**Comment by wez** [comment]
I wonder if this is related to https://github.com/wez/wezterm/issues/3685

**Comment by gegoune** [comment]
Is there anything you would like me to do to test it further? I am still seeing that issue.

**Comment by wez** [comment]
It might be helpful to capture a session recording of the nvim instance in the affected window.
If it is manipulating the palette then it would be resistant to the overall color scheme changing.

* Launch wezterm.  If possible, please use the default terminal size of 80x24 cells as it helps to keep things smaller and easier to manage.
* Inside that terminal run `wezterm record` to start a recording session.
* Run through your reproduction steps: eg: launch nvim
* Then type `exit`
* You should see a message like:

```
*** Finished recording to /var/tmp/wezterm-recording-sF6B3u.cast.txt
```

* Attach the file that it produced to this issue.

The file is an asciicast (compatible with https://asciinema.org/) and can also be replayed using `wezterm replay`.

The terminal recording allows me to replicate what is being sent to the terminal without requiring me to install the same applications as you and replicate your configuration for everything.

**Comment by wez** [comment]
note to self: recording was sent via email with subject "wezterm recordings for #3328"

**Comment by gegoune** [comment]
I can actually reproduce it without neovim. All that's needed on top of configuration from https://github.com/wez/wezterm/issues/3328#issuecomment-1480638385 (with keycaps to switch between workspaces quickly)

```lua
local wezterm = require 'wezterm'

function scheme_for_appearance(appearance)
  if appearance:find 'Dark' then
    return 'Builtin Solarized Dark'
  else
    return 'Builtin Solarized Light'
  end
end

return {
  color_scheme = scheme_for_appearance(wezterm.gui.get_appearance()),
  keys = {
    { key = 'l', mods = 'CTRL|CMD', action = wezterm.action.SwitchWorkspaceRelative(1) },
    { key = 'h', mods = 'CTRL|CMD', action = wezterm.action.SwitchWorkspaceRelative(-1) },
  },
}
```
 is
1. start wezterm
2. open at least one more workspace
3. switch between workspaces, toggle system mode (dark, light) few times and change workspaces between toggles - eventually one of the tabs will not get its colorscheme updated.

(Please note that you might have to switch between modes handful of times before issue appears.)

I am on `wezterm 20231120-164150-fde92672`.

**Comment by 1orZero** [comment]
The Issue still persist

**Comment by S1M0N38** [comment]
I've noticed a related issue: "background" panes do not always update their color scheme.

In this specific case, where I had not run any command or terminal application is running, I was able to update the color scheme with `cmd-R`. But sometimes hitting "cmd-R” does not work.

On macOS:

1. Open wezterm "default" workspace
2. Create a split
3. Create a new workspace and move there (named "handsome-crib")
4. Manually change color scheme
5. Switch back to the "default" workspace.

https://github.com/user-attachments/assets/17bba3e6-e203-4079-aea1-30ee67ddced4

Watching closely, the foreground is correctly updated, but the background is not.

---

I think this could be related to this very issue. Maybe the issue triggering is the same, but due to a long-running application, wezterm falls into a state where it thinks that all the panes have the right color scheme (even though they do not).
